fvm/state/accounts_test.go

```diff
@@ -10,13 +10,14 @@ import (
 )
 
 func TestAccounts_GetWithNoKeys(t *testing.T) {
-	chain := flow.Mainnet.Chain()
-
 	ledger := state.NewMapLedger()
 
-	accounts := state.NewAccounts(ledger, chain)
+	accounts := state.NewAccounts(ledger)
+	addressGenerator := &flow.MonotonicAddressGenerator{}
```

Comment:
Shouldn't the tests use the other `LinearCodeAddressGenerator` as it's the one used in mainnet and testnet ?  `MonotonicAddressGenerator` is just used by the emulator.

Reply:
actually the address generator is not the thing being tested in these tests. and since accounts are now completely decoupled from it, I will just remove it altogether.

Reply:
Updated! 

Also added a new test that will break if any new registers are created during account creation, so that it does not happen unintentionally.